src/lib/common/sol-platform.c

```diff
@@ -394,7 +394,13 @@ sol_platform_get_sw_version(void)
 SOL_API char *
 sol_platform_get_os_version(void)
 {
-    return sol_platform_impl_get_os_version();
+    char *out = NULL;
+
+    int r = sol_util_get_os_version(&out);
```

Comment:
what about us storing this string as a global, returning a `const char *` to that string and free that on `sol_platform_shutdown()`? It's easier to use since people don't need to free() it. And also will cache the results:

```c
static char *os_version;

SOL_API const char *
sol_platform_get_os_version(void)
{
    int r;
    char *out;
    if (os_version) return os_version;
    r = sol_util_get_os_version(&out);
    SOL_INT_CHECK(r, < 0, NULL);
    os_version = out;
    return os_version;
}
```